core/bin/dust_api.rs

```diff
@@ -797,13 +728,15 @@ async fn runs_create_stream(
             });
         }
         Err((_, api_error)) => {
-            let _ = tx.send(json!({
-                "type": "error",
-                "content": {
-                    "code": api_error.code,
-                    "message": api_error.message,
-                },
-            }));
+            if let Some(error) = &api_error.0.error {
```

Comment:
We need a matching branch. I understand it should never be executed but the types don't guarantee that.

I would do 
```
match &api_error.0.error {
  Some(error) => ...
  None => tx.send(...)
```

With code `internal_server_error` and message `The app execution failed unexpectedly` ?